app/add/page.tsx

```diff
@@ -1,140 +1,217 @@
+'use client';
+
+import { FormEvent } from 'react';
 import {
   Container,
   Text,
   TextInput,
+  Textarea,
   Button,
   NumberInput,
   Select,
   Flex,
   Stack,
 } from '@mantine/core';
 import { IconPlus } from '@tabler/icons-react';
+import { useFormData } from '@/hooks/useRecipeFormUpdate';
+import { Unit } from '@/types/recipes';
 
-const AddRecipe = () => (
-  <Container size="sm">
-    <Flex direction="column" gap="16">
-      {/* Header Section */}
-      <Flex justify="space-between" align="center" p="4">
-        <Flex gap="16">
-          <Flex
-            direction="column"
-            align="center"
-            justify="center"
-            bg="#ABEDC6"
-            p="md"
-            w="100px"
-            style={{ borderRadius: '5px', boxShadow: '2px 4px 5px 0px rgba(0, 0, 0, 0.14)' }}
-          >
-            <Text size="sm">Ingredients</Text>
-            <Text size="xl">2</Text>
-          </Flex>
-          <Flex
-            direction="column"
-            align="center"
-            justify="center"
-            bg="#ABEDC6"
-            p="md"
-            w="100px"
-            style={{ borderRadius: '5px', boxShadow: '2px 4px 5px 0px rgba(0, 0, 0, 0.14)' }}
-          >
-            <Text size="sm">Steps</Text>
-            <Text size="xl">1</Text>
-          </Flex>
-        </Flex>
-        <Flex direction="column" gap="8">
-          <Button size="lg" variant="outline" color="pink">
-            Publish
-          </Button>
-          <Button size="lg" variant="outline" color="pink">
-            Save
-          </Button>
-        </Flex>
-      </Flex>
+//TODO: Research best way to: 1. autosave? 2. Keep the form data populated... how can we use jotai to achieve that?
+const AddRecipe = () => {
+  const { formData, updateField, updateIngredient, updateStep } = useFormData();
+
+  const handleSubmit = async (e: FormEvent) => {
```

Comment:
This should re-route the user to the recipes page on success - or clear the form? 

What's better UX? 🤔

Reply:
Maybe just take them right into viewing the recipe - which is what I think I will do for now, and take that chance to build the viewer page. 

The loose design now looks like:

<img width="701" alt="image" src="https://github.com/user-attachments/assets/416fcd65-a019-46fa-9e07-2ff3ed7d4966">

As noted elsewhere, these are rough, basically wireframes of the UI. I don't want to spend too much time designing something before I have a backend system setup to make it all work, and I'm doing that alongside this work.